### ISSUE TITLE
Update Toolbox to solve NARGCHK Deprecated Warning

### ISSUE THREAD

**Issue by pedroperrusi** [pedroperrusi/NLID-matlab-toolbox#1]
The functionality [nargchk](https://www.mathworks.com/help/matlab/ref/nargchk.html) utilized by the original toolbox is now deprecated and shall be removed on future versions of MATLAB.
* obs: MATLAB emits a warning, witch is supressed by the script _supress_warning.m_, called on _setup.m_.

The goal is to verify the toolbox code and update de deprecated function nargchk to [narginchk](https://www.mathworks.com/help/matlab/ref/narginchk.html) or [nargoutchk](https://www.mathworks.com/help/matlab/ref/nargoutchk.html), when applicable.

**Comment by Socratesnote** [comment]
@pedroperrusi , I recently found this repository and plan to use it for my PhD work. I had various problems running the code on Matlab 2018b / Win64, so I did a first cleanup of the code and got it to work. This includes removing the nargchk-calls by appropriate replacements and various other changes that were made necessary by changes to Matlab over the years. Let me know if you are interested in merging my changes into a new branch.

**Comment by pedroperrusi** [comment]
Hello @Socratesnote, its great to know you're working with the toolbox.
A major cleanup and update is indeed essential to work fine on the newer versions of Matlab.

I would love to know what you're up to and update the repository.
Could you prepare a PR? I might be able to review it in a few days.

**Comment by Socratesnote** [comment]
I have a commit ready to be used as a pull-request, but I need write permission to actually publish it to this repo. Or am I missing something?

**Comment by pedroperrusi** [comment]
The standard procedure would be to [create a pull request by a fork](https://help.github.com/articles/creating-a-pull-request-from-a-fork/). 
However, I will be happy to include you as a contributor to the repository. 

Feel welcome to make new contributions and keep the repository updated.
The idea of updating a different branch and then merging it to master might work just fine.

Could you keep me informed of the changes through the issue threads? 
Feel free to open new issues too.

**Comment by pedroperrusi** [comment]
I've sent you an invitation to gain push access to the repository.

**Comment by Socratesnote** [comment]
Closing this issue as the NARGCHK-warning has been fixed on the Win64 branch.

**Comment by Socratesnote** [comment]
Dear Pedro,

Thank you for the access. My apologies for not creating a pull-request 
from a fork: I'm not very familiar with the standard procedures for 
collaboration on Github, since I've only used it for personal repositories.

I have created a new branch, "Win64", to store the changes that I've 
made. I have kept the original "master" branch, since some people might 
rely on the old code for legacy machines. If you want to merge the two, 
feel free.

If I make any more major changes I will create new issue tickets when 
necessary, and update the change logs accordingly.

Thank you for hosting this toolbox, I hope it will come in handy for 
mine and others' work.

Best,

Thomas

-----------------------------------------
Thomas Plaisier
PhD - Biomedical Engineering
Northwestern University, Evanston/Chicago
